### PR TITLE
Allow setting final logout uri

### PR DESCRIPTION
**Pull request by liggitt**
Also move references to global OPENSHIFT_CONFIG into app.js

### CODE REVIEW

**Comment by liggitt** [comment]
@jwforres review

File: assets/test/spec/controllers/project.js [hunk 1]
```diff
@@ -25,9 +25,12 @@ describe('Controller: ProjectController', function () {
     });
   }));
 
+  // Make sure a base location exists in the generated test html
+  if (!$('head base').length) {
+    $('head').append($('<base href="/">'))
+  }
+
   angular.module('openshiftConsole').config(function(AuthServiceProvider) {
-    AuthServiceProvider.LoginService('RedirectLoginService');
```
Comment (jwforres) [inline assets/test/spec/controllers/project.js]: were these duplicated before since we were already setting it in app.js?

Comment (jwforres) [inline assets/test/spec/controllers/project.js]: nvm realized this is the test file

File: pkg/assets/handlers.go [hunk 2]
```diff
@@ -122,11 +123,13 @@ type WebConsoleConfig struct {
...
 	KubernetesPrefix string
 	// OAuthAuthorizeURL is the OAuth2 endpoint to use to request an API token. It must support request_type=token.
-	OAuthAuthorizeURL string
+	OAuthAuthorizeURI string
 	// OAuthRedirectBase is the base URI of the web console. It must be a valid redirect_uri for the OAuthClientID
 	OAuthRedirectBase string
 	// OAuthClientID is the OAuth2 client_id to use to request an API token. It must be authorized to redirect to the web console URL.
 	OAuthClientID string
+	// LogoutURI is an optional (absolute) URI to redirect to after completing a logout. If not specified, the built-in logout page is shown.
+	LogoutURI string
```
Comment (jwforres) [inline pkg/assets/handlers.go]: will this be used by anything other than the web console?

File: pkg/cmd/server/start.go [hunk 3]
```diff
@@ -358,6 +358,8 @@ func start(cfg *config, args []string) error {
 			KubernetesAddr:       cfg.KubernetesAddr.URL.String(),
 			KubernetesPublicAddr: k8sPublicAddr.URL.String(),
 
+			LogoutURI: env("OPENSHIFT_LOGOUT_URI", ""),
```
Comment (jwforres) [inline pkg/cmd/server/start.go]: so we are only allowing this to be set with ENV var right now, no flag for it?

File: pkg/assets/handlers.go (hunk 2 above)
Comment (liggitt) [inline pkg/assets/handlers.go]: not currently... I can't really think of a reason for a non-browser client to use this, but one may come to me

File: pkg/cmd/server/start.go (hunk 3 above)
Comment (liggitt) [inline pkg/cmd/server/start.go]: yeah, all the auth config is env only until we think through how we really want it structured/specified

File: pkg/assets/handlers.go (hunk 2 above)
Comment (jwforres) [inline pkg/assets/handlers.go]: was debating whether it should be ConsoleLogoutURI or not but if we might at some point use it for something else then its fine

**Comment by jwforres** [comment]
LGTM

**Comment by liggitt** [comment]
[merge]

**Pull request merged**